spec/models/page_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe Page, type: :model do
+  context 'markup' do
+    let(:page_with_russian_headers) { FactoryBot.create(:page, source: "# Привет, мир 2!\r\nHello, world!") }
```

Comment:
У меня :source выглядело именно так. Каждый перенос строки шел через "/r/n". Хотя, через простое "/n" тоже работает. В сети пишут, что для винды важна разметка именно в виде "/r/n", мб это для универсальности маркдаун делает?

\r = CR (Carriage Return) → Used as a new line character in Mac OS before X
\n = LF (Line Feed) → Used as a new line character in Unix/Mac OS X
\r\n = CR + LF → Used as a new line character in Windows

Most textual Internet protocols (including HTTP, SMTP, FTP, IRC, and many others) mandate the use of ASCII CR+LF ('\r\n', 0x0D 0x0A) on the protocol level, but recommend that tolerant applications recognize lone LF ('\n', 0x0A) as well. Despite the dictated standard, many applications erroneously use the C newline escape sequence '\n' (LF) instead of the correct combination of carriage return escape and newline escape sequences '\r\n' (CR+LF) (see section Newline in programming languages above). This accidental use of the wrong escape sequences leads to problems when trying to communicate with systems adhering to the stricter interpretation of the standards instead of the suggested tolerant interpretation.

Reply:
С виндой это всё понятно. А можно это какой-то настройкой отключить, чтобы `\n` строго было? Если будет разнообразие вариантов переводов строк — это будет портить жизнь, если внезапно понадобится source руками парсить зачем-то.

Reply:
> С виндой это всё понятно. А можно это какой-то настройкой отключить, чтобы `\n` строго было? Если будет разнообразие вариантов переводов строк — это будет портить жизнь, если внезапно понадобится source руками парсить зачем-то.

gsub можно простой запилить)

А почему у нас будет разнообразие вариантов? Я думал, у нас всегда тут сохраняется перенос строки как /r/n, нет?

Reply:
А нет, вру, простой gsub не получится - у нас "/n" после заголовков переносит строку, а после простого текста - нет. А вот "/r/n" переносит строку в обоих случаях. То есть если jsub`ить, то "/r/n" на "/n/n", а это шило на мыло

Reply:
А, я понял, это просто из браузера они в таком формате передаются. А на выходе уже только `\n`. Я чего-то на беглый взгляд решил, что это в выходном рендере появляется.